### DISCUSSION TITLE
catchall breaks switching to submap if modifier is released before button

### DISCUSSION THREAD

**Discussion by Kommynct** [hyprwm/Hyprland#5032] [Q&A] [answered]
```
#submaps
bind = ALT, a, submap,onehand
bind = SUPER_ALT, 0, submap, "Gaming Mode - Press Alt+shift+0 to exit"
submap=onehand
bind = , e, movefocus, r
bind = ALT,Alt_L,exec,nop
bind = , a, movefocus, l
bind = , COMMA, movefocus, u
bind = , o, movefocus, d
bind = ALT, e, movewindow, r
bind = ALT, a, movewindow, l
bind = ALT, COMMA, movewindow, u
bind = ALT, o, movewindow, d
binde = SUPER,e,resizeactive,10 0
binde = SUPER,a,resizeactive,-10 0
binde = SUPER,COMMA,resizeactive,0 -10
binde = SUPER,o,resizeactive,0 10
bindel =, XF86AudioRaiseVolume, exec, $volup
bindel =, XF86AudioLowerVolume, exec, $voldown
bindl =ALT, UP, exec, $volup
bindl =ALT, DOWN, exec, $voldown
bindl =, XF86AudioMute, exec, $volmute
bindl =, XF86MonBrightnessUp, exec, $lightup
bindl =, XF86MonBrightnessDown, exec, $lightdown
bindl =, XF86AudioPlay, exec, playerctl play
bindl =, XF86AudioPause, exec, playerctl pause
bindl =, XF86AudioNext, exec, playerctl next
bindl =, XF86AudioPrev, exec, playerctl previous
bindm = SUPER, mouse:272, movewindow
bindm = SUPER, mouse:273, resizewindow
bind = ,catchall,submap,reset
submap=reset
```

my current config has this in it, if i push alt+a and release a first, then alt, it works fine, but if i release alt first, it switches away from the submap, i'm using the latest master, any idea how to fix that?

I tried adding bind = ALT,Alt_L,exec,nop and

bind = ,a,exec,nop

to make both of those keys part of the submap always, but it's still exiting it if I release alt first, which is the comfortable way to do it for me.

**Comment by vaxerski** [comment]
have you tried a bindr on alt+a / a / alt with nop?

**Comment by Kommynct** [comment]
Here's what i've tried

`bindr = ALT, a, submap,onehand`

^^this makes it so that it doesn't start at all when I do the rolling motion of alt+a, indicating that something is more fundamentally broken

I've tried bindr = ALT,a,exec,nop within the submap, did not fix it, same result with or without the R

i've already bound just A in the submap so that shouldn't be the issue, but just in case within the submap i tried 

bind = ,a,exec,nop

same result

I've tried bindr of the same thing, same result, i also tried putting both outside the submap but then that just breaks my a key and doesn't solve anything.

Pretty sure this is a bug.

Just to explain the pattern is ALT+A then release alt, then release A, and you'll reproduce it every time on the latest master.

**Reply by vaxerski** [reply, marked as answer]
I mean yea this is a bug, looks like it

**Reply by Kommynct** [reply]
Let me know if there's any way I could help with that, although I know very little

**Reply by vaxerski** [reply]
probably open a ticket
